addons/web/static/src/views/calendar/squareSelectionHook.js

```diff
@@ -0,0 +1,141 @@
+
+            squareSelectionClearState();
+
+            if (mode !== CALENDAR_MODES.filter) {
+                const squareSelectionPointerDownBound = squareSelectionPointerDown.bind(component);
+                const squareSelectionPointerMoveBound = squareSelectionPointerMove.bind(component);
+                const squareSelectionPointerUpBound = squareSelectionPointerUp.bind(component);
+                const squareSelectionPointerCancelBound =
+                    squareSelectionPointerCancel.bind(component);
+                window.addEventListener("pointerdown", squareSelectionPointerDownBound);
+                window.addEventListener("pointermove", squareSelectionPointerMoveBound);
+                window.addEventListener("pointerup", squareSelectionPointerUpBound);
+                window.addEventListener("pointercancel", squareSelectionPointerCancelBound);
+                return () => {
+                    window.removeEventListener("pointerdown", squareSelectionPointerDownBound);
+                    window.removeEventListener("pointermove", squareSelectionPointerMoveBound);
+                    window.removeEventListener("pointerup", squareSelectionPointerUpBound);
+                    window.removeEventListener("pointercancel", squareSelectionPointerCancelBound);
+                };
+            }
+        },
+        () => [component.calendarRef.el, component.props.calendarMode]
+    );
+
+    function getElementIndex(element) {
+        return [].indexOf.call(element?.parentNode.children || [], element);
+    }
+
+    function squareSelectionClearState() {
+        state.startCol = -1;
+        state.endCol = -1;
+        state.startRow = -1;
+        state.endRow = -1;
+
+        state.currentSelectionElement = [];
+    }
+
+    function squareSelectionGetSelectedElement() {
+        const elementsToSelect = [];
+        const [startX, endX] = [state.startCol, state.endCol].sort();
+        const [startY, endY] = [state.startRow, state.endRow].sort();
+
+        for (let x = startX; x <= endX; x++) {
+            for (let y = startY; y <= endY; y++) {
+                elementsToSelect.push(
+                    `tbody tr[role="row"]:nth-child(${y + 1}) > .fc-day:nth-child(${x + 1})`
+                );
+            }
+        }
+
+        if (elementsToSelect.length) {
+            return component.calendarRef.el.querySelectorAll(elementsToSelect.join(","));
+        } else {
+            return [];
+        }
+    }
+
+    function squareSelectionDrawHighlight() {
+        const highlight = "o-highlight";
+
+        component.calendarRef.el.querySelectorAll(`.${highlight}`).forEach((node) => {
+            node.classList.remove(highlight);
+        });
+
+        state.currentSelectionElement.forEach((node) => {
+            node.classList.add(highlight);
+        });
+    }
+
+    function squareSelectionPointerDown(ev) {
+        if (component.props.calendarMode === CALENDAR_MODES.filter) {
```

Comment:
those checks are no longer necessary I think, since you don't add the listeners in that mode